examples/a_treeple/treeple_tutorial_GMM.py

```diff
@@ -0,0 +1,177 @@
+p_x = pdf_class0 * 0.5 + pdf_class1 * 0.5
+
+# %%
+# Calculate true posteriors
+# -------------------------
+#
+# .. math:: \mathbb{P}(Y  \mid X = x) = \frac{f_{X  \mid Y}(x)\mathbb{P}(Y )}{f_{X}(x)}
+
+
+pos_class0 = pdf_class0 * 0.5 / p_x
+pos_class1 = pdf_class1 * 0.5 / p_x
+
+pos = np.hstack((pos_class0.reshape(-1, 1), pos_class1.reshape(-1, 1)))
+
+# %%
+# Generate true statistic estimates: S@98
+# ---------------------------------------
+
+
+def Calculate_SA(y_true, y_pred_proba, max_fpr=0.02) -> float:
+    # check the shape of true labels
+    if y_true.squeeze().ndim != 1:
+        raise ValueError(f"y_true must be 1d, not {y_true.shape}")
+
+    # find the positive class and calculate fpr and tpr
+    if 0 in y_true or -1 in y_true:
+        fpr, tpr, thresholds = roc_curve(
+            y_true, y_pred_proba[:, 1], pos_label=1, drop_intermediate=False
+        )
+    else:
+        fpr, tpr, thresholds = roc_curve(
+            y_true, y_pred_proba[:, 1], pos_label=2, drop_intermediate=False
+        )
+    sa98 = max([tpr for (fpr, tpr) in zip(fpr, tpr) if fpr <= max_fpr])
+    RocCurveDisplay(fpr=fpr, tpr=tpr).plot(label="ROC Curve")
+
+    spec = int((1 - max_fpr) * 100)
+    plt.axvline(
+        x=max_fpr,
+        color="r",
+        ymin=0,
+        ymax=sa98,
+        label="S@" + str(spec) + " = " + str(round(sa98, 2)),
+        linestyle="--",
+    )
+    plt.axhline(y=sa98, xmin=0, xmax=max_fpr, color="r", linestyle="--")
+    plt.legend()
+
+    return sa98
+
+
+sa98 = Calculate_SA(y, pos, max_fpr=0.02)
+
+# %%
+# Generate true statistic estimates: MI
+# -------------------------------------
+
+
+def Calculate_MI(y_true, y_pred_proba):
+    # calculate the conditional entropy
+    H_YX = np.mean(entropy(y_pred_proba, base=np.exp(1), axis=1))
+
+    # empirical count of each class (n_classes)
+    _, counts = np.unique(y_true, return_counts=True)
+    # calculate the entropy of labels
+    H_Y = entropy(counts, base=np.exp(1))
+    return H_Y - H_YX
+
+
+mi = Calculate_MI(y, pos)
+print("MI =", round(mi, 2))
+
+
+# %%
+# Generate true statistic estimates: pAUC@90
+# ------------------------------------------
+
+
+def Calculate_pAUC(y_true, y_pred_proba, max_fpr=0.1) -> float:
+    # check the shape of true labels
+    if y_true.squeeze().ndim != 1:
+        raise ValueError(f"y_true must be 1d, not {y_true.shape}")
+
+    # find the positive class and calculate fpr and tpr
+    if 0 in y_true or -1 in y_true:
+        fpr, tpr, thresholds = roc_curve(
+            y_true, y_pred_proba[:, 1], pos_label=1, drop_intermediate=False
+        )
+    else:
+        fpr, tpr, thresholds = roc_curve(
+            y_true, y_pred_proba[:, 1], pos_label=2, drop_intermediate=False
+        )
+    RocCurveDisplay(fpr=fpr, tpr=tpr).plot(label="ROC Curve")
+    # Calculate pAUC at the specific threshold
+    pAUC = roc_auc_score(y_true, y_pred_proba[:, 1], max_fpr=max_fpr)
+
+    pos = np.where(fpr == max_fpr)[0][-1]
+    plt.fill_between(
```

Comment:
Do we want to separate plotting and obtaining the statistic? Makes it a bit cleaner to understand from a user perspective